### PR TITLE
oo testing: skip building index images when file-based index is used

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-2969

/cc @openshift/test-platform @joelanford @grokspawn 

/hold

This should not go in before the PR on the ci-docs is ready.

### CODE REVIEW

File: pkg/steps/index_generator.go [hunk 1]
```diff
@@ -63,6 +62,7 @@ func (s *indexGeneratorStep) Run(ctx context.Context) error {
 }
 
 func (s *indexGeneratorStep) run(ctx context.Context) error {
+	logrus.Warn("Building index images is deprecated and will be removed from ci-operator soon")
```
Comment (smg247) [inline pkg/steps/index_generator.go:65]: Maybe this needs to be a little more noticeable:
```suggestion
	logrus.Warn("DEPRECATION WARNING: Building index images is deprecated and will be removed from ci-operator soon")
```

File: pkg/steps/index_generator.go [hunk 2]
```diff
@@ -63,6 +62,7 @@ func (s *indexGeneratorStep) Run(ctx context.Context) error {
 }
 
 func (s *indexGeneratorStep) run(ctx context.Context) error {
+	logrus.Warn("DEPRECATION WARNING: Building index images is deprecated and will be removed from ci-operator soon")
```
Comment (hongkailiu) [inline pkg/steps/index_generator.go:65]:
```suggestion
	logrus.Warn("DEPRECATION WARNING: Building index images is deprecated and will be removed from ci-operator soon. See https://docs.ci.openshift.org/docs/how-tos/testing-operator-sdk-operators/#moving-to-file-based-catalog for details.")
```

**Pull request merged**